### PR TITLE
fix(kubernetes-actions): fix code smells

### PR DESCRIPTION
**Pull request by anilabhabaral**
Resolves: [#592](https://github.com/janus-idp/backstage-plugins/issues/592)

### CODE REVIEW

File: plugins/kubernetes-actions/src/actions/createKubernetesNamespace.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ const getUrlFromClusterRef = async (
 const validateUrl = (url: string | undefined) => {
   try {
     // eslint-disable-next-line no-new
-    new URL(url!);
+    new URL(url as string);
```
Comment (tumido) [inline plugins/kubernetes-actions/src/actions/createKubernetesNamespace.ts:71]: Can you please create a test case for this? https://github.com/janus-idp/backstage-plugins/blob/main/plugins/kubernetes-actions/src/actions/createKubernetesNamespace.test.ts

I'm not sure casting this to string is the best option. What about:
```suggestion
    new URL(url || "");
```

or even better, in the signature:

```diff
- const validateUrl = (url: string | undefined) => {
+ const validateUrl = (url: string | undefined = "") => {
```

**Review by tumido — CHANGES_REQUESTED** [review]

File: plugins/kubernetes-actions/src/actions/createKubernetesNamespace.ts (hunk 1 above)
Comment (anilabhabaral) [reply plugins/kubernetes-actions/src/actions/createKubernetesNamespace.ts:71]: Thanks for the suggeation @tumido. I believe `const validateUrl = (url: string | undefined = "") => {` would be best for this use case as it is by default assigning the empty string to undefined and we don't need to check/set the type of `url`.

Comment (anilabhabaral) [reply plugins/kubernetes-actions/src/actions/createKubernetesNamespace.ts:71]: I will also write the tests for empty string `""` and the `undefended` value to clarify the above.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Comment by anilabhabaral** [comment]
Thank You @tumido for your help!
